connectors/webhook/src/main/java/io/camunda/connector/inbound/signature/strategy/URLAndBodyEncodingStrategy.java

```diff
@@ -15,30 +15,45 @@
 import java.util.Map;
 
 public final class URLAndBodyEncodingStrategy implements HMACEncodingStrategy {
-  @Override
-  public byte[] getBytesToSign(final WebhookProcessingPayload payload) throws IOException {
-    return (payload.requestURL() + extractSignatureData(payload)).getBytes();
-  }
 
-  private static String extractSignatureData(final WebhookProcessingPayload payload)
-      throws IOException {
+  private static String extractSignatureData(final WebhookProcessingPayload payload) {
     if (payload.rawBody() == null || payload.rawBody().length == 0) {
       throw new NullPointerException(
           "Can't extract signature data from body, because body is null");
     }
     Map<String, String> signatureData =
-        HttpWebhookUtil.transformRawBodyToMap(
-            payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers()));
+        checkedCastToMap(
+            HttpWebhookUtil.transformRawBodyToObject(
+                payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers())),
+            String.class);
 
-    StringBuilder builder = new StringBuilder();
     List<String> sortedKeys = new ArrayList<>(signatureData.keySet());
     Collections.sort(sortedKeys);
 
+    StringBuilder builder = new StringBuilder();
+
     for (String key : sortedKeys) {
       builder.append(key);
       String value = signatureData.get(key);
       builder.append(value == null ? "" : value);
     }
     return builder.toString();
   }
+
+  private static <T> Map<T, T> checkedCastToMap(Object o, Class<T> tClass) {
+    if (o instanceof Map<?, ?> map) {
+      for (Map.Entry<?, ?> entry : map.entrySet()) {
+        if (!(tClass.isInstance(entry.getValue())) || !(tClass.isInstance(entry.getKey()))) {
+          return Map.of();
+        }
+      }
+      return (Map<T, T>) map;
+    }
+    return Map.of();
+  }
```

Comment:
As we cant calculate the signature of an empty map anyway. What about skipping it when the payload is not of type map?

Reply:
If we have a `String[]` couldn't we use the values to calculate the signature? 
I don't know where it is used, so I might be wrong

Reply:
Not send as an array, but we could build the signature in a similar way of what we do with Maps, basically concatenating values

Reply:
So the given list would be in this format `["key1", "value1", "key2", "value2"]` @johnBgood ?

Reply:
Yes exactly :) Conceptually it would be more `["value1", "value2"...]` but that's the spirit yes.
Instead of adding keys/values from a Map (see below), we would simply add the array's values.

```java
for (String key : sortedKeys) {
      builder.append(key);
      String value = signatureData.get(key);
      builder.append(value == null ? "" : value);
    }
```